Library/Homebrew/cask/dsl.rb

```diff
@@ -92,6 +95,30 @@ def name(*args)
       @name.concat(args.flatten)
     end
 
+    def license(*args)
+      @licenses ||= []
+
+      return @licenses unless args
+
+      valid_licenses = Licensee.licenses.map(&:key)
```

Comment:
I think it's worth just allowing any value of license to be specified and relying on `brew audit` and `brew cask audit` to flag invalid ones.